daily-challenges/2024-09/17/tunganh222Z.java

```diff
@@ -0,0 +1,122 @@
+    By product3Table = By.xpath("//td[text()='Product 3']/parent::tr");
+
+    By quantity1 = By.xpath("//td[text()='Product 1']/parent::tr/td[3]");
+    By quantity2 = By.xpath("//td[text()='Product 2']/parent::tr/td[3]");
+    By quantity3 = By.xpath("//td[text()='Product 3']/parent::tr/td[3]");
+
+    @BeforeClass
+    public void beforeClass(){
+        driver = new FirefoxDriver();
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(15));
+        explicitWait = new WebDriverWait(driver, Duration.ofSeconds(15));
+        driver.manage().window().maximize();
+    }
+
+    @Test
+    public void TC_01_prime(){
+        //dataTest_TC_01
+        int number = 23;
+        //
+        System.out.println(isPrime(number));
+    }
+
+    @Test
+    public void TC_02_productPage(){
+        //Open
+        driver.get("https://material.playwrightvn.com/");
+        driver.findElement(By.xpath("//a[contains(text(),'Product page')]")).click();
+
+        //dataTest_TC_02
+        int expectedQuantity1 = 2;
+        int expectedQuantity2 = 2;
+        int expectedQuantity3 = 3;
+
+        //Add to cart product1
+        webElement(product1).click();
+        addToCart(product1Table, webElement(product1), webElement(quantity1), expectedQuantity1);
+
+        //Add to cart product2
+        webElement(product2).click();
+        addToCart(product2Table, webElement(product2), webElement(quantity2), expectedQuantity2);
+
+        //Add to cart product3
+        webElement(product3).click();
+        addToCart(product3Table, webElement(product3), webElement(quantity3), expectedQuantity3);
+
+        //Check total price
+        double totalPriceD = totalPrices(webElement(price1),expectedQuantity1) + totalPrices(webElement(price2),expectedQuantity2) + totalPrices(webElement(price3),expectedQuantity3);
+        System.out.println("Total price is : $"+totalPriceD);
+        //convert to Strring
+        String totalPriceS = "$"+String.format("%.2f", totalPriceD);
+        //Assert
+        Assert.assertEquals(driver.findElement(By.xpath("//td[@class='total-price']")).getText(),totalPriceS);
+    }
+
+    public double totalPrices(WebElement price, int expectedQuantity){
+        //
+        double priceString = Double.parseDouble(price.getText().replace("$",""));
+        double totalPrice = priceString * expectedQuantity;
+        return totalPrice;
+    }
+
+    public WebElement webElement(By locator){
+        explicitWait.until(ExpectedConditions.visibilityOfElementLocated(locator));
+        return driver.findElement(locator);
+    }
+
+    public void  addToCart(By productTable, WebElement product, WebElement quantity, int expectedQuantity){
+        explicitWait.until(ExpectedConditions.visibilityOfElementLocated(productTable));
+        if (driver.findElement(productTable).isDisplayed()){
+            for (int i = 1; i < expectedQuantity ; i++){
+                if (!quantity.equals(expectedQuantity)){
```

Comment:
Đây cũng là một cách. Tuy nhiên code này làm phức tạp test case.
Just keep it simple: add 2 lần thì click 2 lần thôi.
Có thể viết một hàm, truyền vào locator và click count. Trong đó thì click thôi là được.

Reply:
Vâng ạ, vì em vừa muốn mỗi lần click sẽ verify luôn quantity nên viết ra hàm như này ạ, lần sau e sẽ sửa đổi lại